UPGRADE-4.2.md

```diff
@@ -1,6 +1,27 @@
 UPGRADE FROM 4.1 to 4.2
 =======================
 
+Form
+----
+
+ * Deprecated calling `FormRenderer::searchAndRenderBlock` for fields which were already rendered. 
+   Instead of expecting such calls to return empty string, check if the field has already been rendered.
+ 
+   Before:
+   ```twig
+   {% for field in fieldsWithPotentialDuplicates %}
+      {{ form_widget(field) }}
+   {% endfor %}
+   ```
+   
+   After:
+   ```twig
+   {% for field in fieldsWithPotentialDuplicates %}
+     {% if not field.rendered %}
```

Comment:
can be simplified: `{% for field in fieldsWithPotentialDuplicates if not field.rendered %}`

Reply:
IIRC this kind of filtering has limitations, depending on what's inside the for block, so I went with safer approach here. But I don't remember what was the issue then, so will change.

Reply:
the limitation is that any `loop.*` variables depending on the number of items in the loop won't work anymore (because it won't know in advance).
But if you skip the rendering entirely for some items with an inner filtering, these would not work well anyway, as some code using `loop.last` would probably expect it to be the last *rendered* item, not the last checked one.

Reply:
then it is not the same original code (and then, you would have to evaluate whether the additional code should run when the field does not get rendered), and so requires a case-by-case decision.